elide-datastore/elide-datastore-hibernate5/src/main/java/com/yahoo/elide/datastores/hibernate5/HibernateTransaction.java

```diff
@@ -123,15 +122,61 @@ public void commit() {
         criterion = CriterionFilterOperation.andWithNull(criterion,
                 criterionFilterOperation.applyAll(filteredPredicates));
 
-        // if no criterion then return all objects
-        if (criterion == null) {
-            return loadObjects(loadClass);
+
+        final Pagination pagination = filterScope.hasPagination() ? filterScope.getRequestScope().getPagination()
+                : null;
+
+        // if we have sorting and sorting isn't empty, then we should pull dictionary to validate the sorting rules
+        Set<Order> validatedSortingRules = null;
+        if (filterScope.hasSortingRules()) {
+            final Sorting sorting = filterScope.getRequestScope().getSorting();
+            final Map<String, Sorting.SortOrder> validSortingRules = sorting.getValidSortingRules(loadClass,
+                    filterScope.getRequestScope().getDictionary());
+            if (!validSortingRules.isEmpty()) {
```

Comment:
I would think if filterScope.hasSortingRules is true, validSortingRules should not be empty.  I guess if we throw an exception when we parse the sort query params, we won't need this extra check here.